libethereum/EthereumHost.cpp

```diff
@@ -164,7 +163,7 @@ class EthereumPeerObserver: public EthereumPeerObserverFace
 	}
 
 private:
-	BlockChainSync& m_sync;
+	shared_ptr<BlockChainSync> m_sync;
```

Comment:
I'm afraid we have the same problem with `m_syncMutex` below. If `EthereumHost` gets destroyed before `EthereumPeerObserver` mutex is destroyed with it... Not sure yet what to do with it

`m_tq` is destroyed with the `Client`, also not sure if `EthereumPeers` can outlive `Client`

Reply:
(for myself) try to remove `m_syncMutex`.

Reply:
Somehow I'm making other classes hold `m_tq` as a weak pointer.

Reply:
And I was able to remove `m_syncMutex` and still seeing no warnings from ThreadSanitizer.

Reply:
About `m_tq`, I decided to use shared/weak pointers.